proposals/0000-swift-settings.md

```diff
@@ -0,0 +1,446 @@
+# #SwiftSettings: a macro for enabling compiler flags locally in a specific file
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Michael Gottesman](https://github.com/gottesmm)
+* Review Manager: TBD
+* Status: Implemented on main behind flag -enable-experimental-feature SwiftSettings
+* Vision: [[Prospective Vision] Improving the approachability of data-race safety](https://forums.swift.org/t/prospective-vision-improving-the-approachability-of-data-race-safety/76183)
+* Previous Revision: [1](https://github.com/swiftlang/swift-evolution/blob/dbab768bc058c23b7d5b99827680a19443519136/proposals/0000-compiler-settings.md) [2](https://github.com/swiftlang/swift-evolution/blob/3029caa010ee238d8396c1a5014a21b865c5ecb0/proposals/0000-swift-settings.md)
+* Review: ([pitch](https://forums.swift.org/t/pitch-compilersettings-a-top-level-statement-for-enabling-compiler-flags-locally-in-a-specific-file/))
+
+## Introduction
+
+We propose introducing a new macro called `#SwiftSettings` that allows for
+compiler flags to be enabled on a single file. Initially this will only be used
+to support controlling the default isolation. For example:
+
+```swift
+#SwiftSettings(
+    .defaultIsolation(MainActor.self)
+)
+```
+
+In the future, we wish to extend this to also include support for controlling
+`warningsAsErrors`, `strictConcurrency`, and other flags. See future directions
+for more details.
+
+## Motivation
+
+Today Swift contains multiple ways of changing compiler and language behavior
+from the command line including specifying the default actor isolation via
+[Controlling Default Actor Isolation](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0466-control-default-actor-isolation.md). Since
+these compiler and language behaviors can only be manipulated via the command
+line and the compiler compiles one module at a time, users are forced to apply
+these flags one module at a time instead of one file at a time. When confronted
+with this, users are forced to split the subset of files into a separate module
+creating an unnecessary module split. We foresee that this will occur more often
+in the future if new features like [Controlling Default Actor Isolation](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0466-control-default-actor-isolation.md)
```

Comment:
I think this specific argument isn't compelling, because we really don't want to introduce more features like default actor isolation in the future. I think you can just remove this sentence and say this applies for default actor isolation and also for the various diagnostic control flags.